influxdb3_server/src/http.rs

```
@@ -1383,7 +1385,36 @@ fn validate_auth_header(header: HeaderValue) -> Result<Vec<u8>, AuthenticationEr
         return Err(AuthenticationError::MalformedRequest);
     }
 
-    Ok(token.as_bytes().to_vec())
+    let token = if auth_scheme == "Basic" {
+        token_part_as_bytes(token)?
+    } else {
+        token.as_bytes().to_vec()
+    };
+
+    Ok(token)
+}
+
+fn token_part_as_bytes(token: &str) -> Result<Vec<u8>, AuthenticationError> {
+    let decoded = B64_STANDARD.decode(token).map_err(|err| {
+        error!(?err, "cannot decode basic auth token");
+        AuthenticationError::MalformedRequest
+    })?;
+
+    let token_parts = String::from_utf8(decoded)
+        .map_err(|err| {
+            error!(?err, "cannot decode basic auth token to string");
+            AuthenticationError::MalformedRequest
+        })?
+        .split(":")
+        .map(|part| part.to_owned())
+        .collect::<Vec<String>>();
+
+    let token_part = token_parts.get(1).ok_or_else(|| {
```

This is a bit of a nitpick but this will not work right if the user happens to specify `:` in the username (eg, `foo:bar:apiv3_...`). Granted, username is ignored and a malformed username like this will cause the token to be `bar` which will result in a 401 so this is ok from a security POV (failing closed), but it seems we should a) ensure there is at least 1 `:`, b) split on `:` and c) take the last part, not `1`. Perhaps use `.last()` instead of `.get(1)` (and be sure the length of the vector is 2 or more)?

While I don't foresee us ever considering the username, if we ever did want to, adjusting this function would avoid a latent bug.

---

Might be a case worthy of [`str::rsplit_once`](https://doc.rust-lang.org/std/primitive.str.html#method.rsplit_once) so that for, e.g., `foo:bar:apiv3_...`, it will only take `apiv3_...`.

---

> I could be misinterpreting the [RFC](https://datatracker.ietf.org/doc/html/rfc7617), I thought `:` is not allowed in username?
> 
> > For the user-id, recipients MUST support all characters defined in
> > the "UsernameCasePreserved" profile defined in Section 3.3 of
> > [RFC7613], with the exception of the colon (":") character.
> 
> I'll still change this implementation but I interpreted that RFC section as clients won't be able to set ":" in username part of the `user-pass` pair.

Hmm, you're right: "_Furthermore, a user-id containing a colon character is invalid, as the first colon in a user-pass string separates user-id and password from one another; text after the first colon is part of the password. User-ids containing colons cannot be encoded in user-pass strings._"

With that in mind, like @hiltontj suggested, you can a) ensure that `:` is present and b) split once and use `get(1)` (or `last()`). The end result is quite similar to your original implementation but is explicitly coded according to the RFC (so even more of a nitpick, heh).